engines/ags/engine/ac/string.cpp

```diff
@@ -325,11 +326,7 @@ buffr[199]=0;
 void my_strncpy(char *dest, const char *src, int len) {
 	// the normal strncpy pads out the string with zeros up to the
 	// max length -- we don't want that
-	if (strlen(src) >= (unsigned)len) {
-		strncpy(dest, src, len);
-		dest[len] = 0;
-	} else
-		strcpy(dest, src);
+	Common::strlcpy(dest, src, len);
```

Comment:
```c++
	// This function takes the length of the buffer minus 1 (i.e. the length does not include the null terminator).
	Common::strlcpy(dest, src, len + 1);
````

You can see it in the old implementation `dest[len] = 0;` as well as in places where it is called (e.g. `my_strncpy(destt, text, _G(MAXSTRLEN) - 1);`).

Reply:
Thanks! I will fix it when I have time to continue the PR.

Reply:
Done.